### DISCUSSION TITLE
如何编译CMakeLists不在项目根目录下的库呢

### DISCUSSION THREAD

**Discussion by druidfund** [xmake-io/xmake#4713] [General]
xmake-io自带的lz4库不是使用cmake编译，所以没有生成*.cmake文件，而在编译arraow的时候需要*.cmake文件来找到lz4库的位置，我看了一下Lz4库代码，CMakeLists 文件在代码根目录下的build/cmake下，导致xmake.lua里面报错，在cacheddir下面找不到cmakelists，这种情况如何处理呢

**Comment by star-hengxing** [comment]
1. 把该库 port 到 xmake
2. lz4 改回 cmake 编译，或用 xmake 内置规则导出 .cmake，看能不能用
3. 尝试 patch cmakelists，让他能找到该库；或者不让他找库，直接通过 packagedeps 来编译

**Comment by waruqi** [comment]
可以先尝试在 lz4/xmake.lua 中加 cmake export rules ，导出 .cmake 文件，参考其他包
